### ISSUE TITLE
Ability to add a specific revision from a local git repository

### ISSUE THREAD

**Issue by kalvenschraut** [getsentry/sentry-cli#1449]
My repository is a monorepo on local gitea instance we are self hosting. I want to be able to specify the specific commits that effect the specific project the release is being created for because not all commits in the tree may have affected the specific project/app. 

I can pretty trivially figure out what commits are relevant, but doesn't seem to be a way to  tell set-commit to just mark a specific revision as part of the release.

**Comment by kamilogorek** [comment]
https://docs.sentry.io/product/cli/releases/#sentry-cli-commit-integration

> In case you are deploying without access to the git repository, you can manually specify the commits instead. To do this, pass the --commit parameter to the set-commits command in the format REPO_NAME@REVISION. You can repeat this for as many repositories as you have:

Isn't this what you are looking for?

**Comment by kalvenschraut** [comment]
We don't have a repo based integration, unless I am mistake you don't integrate with a locally hosted [gitea](https://github.com/go-gitea/gitea)

I tried doing `ORIGIN@REVISION` since that came up in the repo list command but my commits are not associated to the release.

![image](https://user-images.githubusercontent.com/30308012/216053170-18c0e742-8a0b-4e0a-85ee-09cfe0fed059.png)
above is using the --auto flag. It is grabbing commits that actually didn't affect this specific project since the repo is a monorepo

![image](https://user-images.githubusercontent.com/30308012/216053305-52d389e5-4077-4f55-8720-734f9c3dde89.png)
above is my attempt at use `ORIGIN@REVISION` it did output the below but it didn't attach to the release. I want to be able to just associate specific revisions in my local repo to the release.
![image](https://user-images.githubusercontent.com/30308012/216053430-991f902a-dadd-4770-bc63-b6ff4aaacb53.png)

**Comment by kalvenschraut** [comment]
Is this something that can be done or looked into in the future? As is I can't correctly associate commits with releases

**Comment by szokeasaurusrex** [comment]
@kalvenschraut Indeed, it appears that the `-c` option does not currently support local git repos. We can look into adding support for this.

Although, I also don't think there is any harm in having the release contain commits that are unrelated to your specific project. Unless there is some reason why this would not work for your use case, it might be simpler to just have a single Sentry release for the entire monorepo containing all the commits since the last release, and you could associate this release with all of the projects in the monorepo (releases should be able to be associated with multiple projects)

**Comment by kalvenschraut** [comment]
We currently do project specific versioning in the monorepo and only bump a project version used for the release if a change occurred to it since the prior release. If I were to do that direction might have to be some date/time based release in sentry that includes all changes for all projects that went out in that moment. About to add a few new projects we are about to do the first prod release on next week to sentry and can take a look at what the looks like then.

**Comment by szokeasaurusrex** [comment]
Ah okay, I get it @kalvenschraut – in that case, my suggestion might not be helpful.

We should investigate adding support to the Sentry CLI for this use case. I think it should be possible to do
